src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -640,5 +712,93 @@ private NamedTypeSymbol MakeAcyclicBaseType(DiagnosticBag diagnostics)
+                if (extendedTypeList == null)
+                {
+                    continue;
+                }
+                var extendedTypeBinder = this.DeclaringCompilation.GetBinder(extendedTypeList);
+                foreach (var extendedTypeSyntax in extendedTypeList.Types)
+                {
+                    var typeSyntax = extendedTypeSyntax.Type;
+                    var newExtendedType = extendedTypeBinder.BindType(typeSyntax, diagnostics, null);
+                    if ((object)extendedType != null)
+                    {
+                        Debug.Assert((object)multipleTypesSpecified == null);
+                        multipleTypesSpecified = ArrayBuilder<TypeSymbol>.GetInstance();
+                        multipleTypesSpecified.Add(extendedType);
+                        extendedType = null;
+                    }
+                    if ((object)multipleTypesSpecified != null)
+                    {
+                        multipleTypesSpecified.Add(newExtendedType);
+                    }
+                    else if ((object)extendedType == null)
+                    {
+                        extendedType = newExtendedType;
+                    }
+                }
+            }
+
+            if ((object)multipleTypesSpecified != null)
+            {
+                // PROTOTYPE: Temporary error message just to return some form of an ErrorTypeSymbol
+                var symbols = multipleTypesSpecified.ToImmutableAndFree();
+                var info = new CSDiagnosticInfo(ErrorCode.ERR_AmbigMember, symbols.CastArray<Symbol>(),
+                    new object[] { symbols[0], symbols[1] });
+                return new ExtendedErrorTypeSymbol(this, symbols.CastArray<Symbol>(), LookupResultKind.Ambiguous, info, 0);
+            }
+
+            if ((object)extendedType == null)
+            {
+                // PROTOTYPE: Temporary error message just to return some form of an ErrorTypeSymbol
+                var info = new CSDiagnosticInfo(ErrorCode.ERR_NameNotInContext, this);
+                return new ExtendedErrorTypeSymbol(this, LookupResultKind.Empty, info);
+            }
+
+            var extendedNamedType = extendedType as NamedTypeSymbol;
+            if ((object)extendedNamedType != null && extendedNamedType.IsExtensionClass)
+            {
+                // PROTOTYPE: Temporary error message just to return some form of an ErrorTypeSymbol
+                var info = new CSDiagnosticInfo(ErrorCode.ERR_NameNotInContext, this);
+                return new ExtendedErrorTypeSymbol(extendedType, LookupResultKind.Empty, info);
+            }
+
+            // note no cycles can occur, we are static (is this true? The other class might be an error but it still resolves)
+            this.SetKnownToHaveNoDeclaredBaseCycles();
+
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            if (!this.IsNoMoreVisibleThan(extendedNamedType, ref useSiteDiagnostics))
```

Comment:
`extendedNamedType` may be null. Should this be `extendedType` instead?

Consider changing first use of `extendedNamedType` to `if ((extendedType as NamedTypeSymbol)?.IsExtensionClass)` and remove `extendedNamedType` completely.